api/envoy/extensions/filters/http/oauth2/v3/oauth.proto

```diff
@@ -0,0 +1,74 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.http.oauth2.v3;
```

Comment:
Based on the review iterations, I'm inclining to make this v3alpha with `work_in_progress` annotation. So we are flexible to make breaking change in future to make the filter more generic. WDYT?

Reply:
Yep that makes sense to me